src/Symfony/Component/Notifier/Transport/Dsn.php

```diff
@@ -96,6 +97,15 @@ public function getOption(string $key, $default = null)
         return $this->options[$key] ?? $default;
     }
 
+    public function getRequiredOption(string $key)
+    {
+        if (!\array_key_exists($key, $this->options) || empty($this->options[$key])) {
+            throw new RequiredOptionMissingException($key);
```

Comment:
Maybe: `throw new RequiredOptionMissingException($key, $this->getOriginalDsn());` ?

Reply:
You are right 👍  Thanks

Reply:
Oh I've done that, but we have a "bigger" problem here.

https://github.com/symfony/symfony/blob/026da8b58ecc8171e9c4301e4e6a726595208af3/src/Symfony/Component/Notifier/Transport/Dsn.php#L104-L107

should return `string`, but it contains `null` whenever you instantiate the `Dsn` class vie `new` instead of `fromString()`

https://github.com/symfony/symfony/blob/026da8b58ecc8171e9c4301e4e6a726595208af3/src/Symfony/Component/Notifier/Transport/Dsn.php#L63-L64

One option could be to change the return type to `?string` or make the constructor private and only instantiate through `fromString`. Currently `new Dsn(...)` is only used in tests. 🤔

Reply:
I resolve this, but created a new issue: https://github.com/symfony/symfony/issues/39533